liblocation/src/main/java/com/mapbox/services/android/core/permissions/ExponentialBackoff.java

```diff
@@ -0,0 +1,118 @@
+package com.mapbox.services.android.core.permissions;
+
+/**
+ * Simplified version of Google HTTP Client Library for Java's ExponentialBackOff.
+ * https://developers.google.com/api-client-library/java/google-http-java-client/backoff
+ *
+ * In this version, the interval creation is deterministic and has no concept of
+ * maxElapsedTimeMillis, it's up to the application to stop. It also adjusts its default values.
+ */
+public class ExponentialBackoff {
```

Comment:
This class is only used internally, so no need to be `public` (we don't need to expose it publicly). The same thing happens with the properties and the methods included.